cclib/method/ddec.py

```diff
@@ -271,59 +281,105 @@ def calculate(self, indices=None, fupdate=0.05):
             self.radial_grid_r.append(r)
 
         # Carry out step 3 of DDEC6 algorithm [Determine conditioned charge density and tau]
-        self.logger.info("Conditioning charge densities.")
+        self.logger.info("Conditioning charge densities. (Step 3/7)")
         self.condition_densities()
 
-        # STEP 4-7
-        # Optimize weights on each grid point
-        kappa = [0.0 for x in self.data.atomnos]
+        # Steps 4 through 7 contain similar routines. Comments the precede each step explain the
+        # differences among them.
+
+        # * STEP 4 *
+        # In step 4, calculate w, u, g, and h but always skip kappa updates.
+        self._kappa = [0.0] * self.data.atomnos
         update_kappa = False
+        # N_A is assigned number of electrons determined using equation 72.
         self.N_A = []
-        steps = 4
-        while steps < 5:  # change this to 8 after implementing kappa routine
-            self.logger.info("Optimizing charge weights (Step {}).".format(steps))
+        # u_A is determined using equation 77.
+        self.u_A = []
+
+        self.logger.info("Optimizing grid weights. (Step 4/7)")
```

Comment:
It would help the readability if you placed these logging calls right below the section comments.

Reply:
I've moved the log calls to right below the section comment. Thanks.